mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -148,12 +156,36 @@ def restore_run(self, run_id: str) -> None:
         self._check_run_is_deleted(run)
         run.update(lifecycle_stage=LifecycleStage.ACTIVE)
 
+    @staticmethod
+    def _update_latest_metric_if_necessary(new_metric: ElasticMetric, run: ElasticRun) -> None:
+        def _compare_metrics(metric_a: ElasticLatestMetric, metric_b: ElasticLatestMetric) -> bool:
+            return (metric_a.step, metric_a.timestamp, metric_a.value) > \
+                   (metric_b.step, metric_b.timestamp, metric_b.value)
+        new_latest_metric = ElasticLatestMetric(key=new_metric.key,
+                                                value=new_metric.value,
+                                                timestamp=new_metric.timestamp,
+                                                step=new_metric.step)
+        latest_metric_exist = False
+        for i, latest_metric in enumerate(run.latest_metrics):
+            if latest_metric.key == new_metric.key:
+                latest_metric_exist = True
+                if _compare_metrics(new_latest_metric, latest_metric):
+                    run.latest_metrics[i] = new_latest_metric
+        if not (latest_metric_exist):
+            run.latest_metrics.append(new_latest_metric)
+
     def log_metric(self, run_id: str, metric: Metric) -> None:
+        is_nan = math.isnan(metric.value)
+        if is_nan:
+            value = 0
+        else:
+            value = metric.value
         run = self._get_run(run_id=run_id)
         new_metric = ElasticMetric(key=metric.key,
-                                   value=metric.value,
+                                   value=value,
                                    timestamp=metric.timestamp,
                                    step=metric.step)
```

Comment:
is_nan is also a field of the metric, to know if the 0 we inserted is really a nan or an actual 0, so you need to add this field to your object